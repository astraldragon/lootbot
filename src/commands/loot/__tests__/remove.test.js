jest.mock("discord-akairo");

jest.mock("../../../models", () => ({
  Loot: { destroy: jest.fn("destroy") }
}));

const { Loot } = require("../../../models");
const { Command } = require("discord-akairo");
const { checkManagePermissions } = require("../../../support");
const RemoveCommand = require("../remove");

describe("commands/loot/remove", () => {
  beforeEach(() => {
    this.name = "Loot 1";
    this.command = new RemoveCommand();

    this.msg = {
      guild: { id: "1234" },
      channel: {
        send: jest.fn("send").mockReturnValue(Promise.resolve())
      }
    };
  });

  it("should configure properly", () => {
    expect(Command).toMatchSnapshot();
  });

  it("should have the manage permission check", () => {
    expect(Command.mock.calls[0][1].options.permissions).toBe(
      checkManagePermissions
    );
  });

  it("should destroy the passed in loot", async () => {
    Loot.destroy.mockReturnValue(1);
    await this.command.exec(this.msg, { name: this.name });

    expect(Loot.destroy).toHaveBeenCalledWith({
      where: { name: this.name, guild: this.msg.guild.id }
    });

    expect(this.msg.channel.send).toHaveBeenCalledWith(`${this.name} removed`);
  });

  it("should notify when the passed in loot does not exist", async () => {
    Loot.destroy.mockReturnValue(0);
    await this.command.exec(this.msg, { name: this.name });

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `${this.name} not found`
    );
  });

  it("should notify if any errors occur", async () => {
    Loot.destroy.mockImplementation(() => {
      throw new Error("test");
    });

    await this.command.exec(this.msg, { name: this.name });

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `An error occurred removing ${this.name}`
    );
  });
});
