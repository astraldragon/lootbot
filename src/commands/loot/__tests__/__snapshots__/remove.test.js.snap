// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/loot/remove should configure properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "loot-remove",
      Object {
        "aliases": Array [
          "loot-remove",
          "lr",
        ],
        "args": Array [
          Object {
            "id": "name",
            "prompt": Object {
              "start": "What is the name of the loot?",
            },
            "type": "string",
          },
        ],
        "category": "Loot",
        "channelRestriction": "guild",
        "description": Object {
          "content": "Remove that loot",
          "examples": Array [
            "loot-remove \\"Maple Syrup\\"",
            "loot-remove Syrup",
          ],
          "usage": "<name>",
        },
        "options": Object {
          "permissions": [Function],
        },
      },
    ],
  ],
}
`;
